experimental/segmenter/src/lstm_structs.rs

```diff
@@ -2,53 +2,90 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use crate::lstm_error::Error;
+
 use alloc::borrow::Cow;
 use icu_provider::prelude::*;
-use ndarray::Array1;
-use ndarray::Array2;
-use serde::{Deserialize, Serialize};
-use zerovec::ZeroMap;
+use ndarray::{Array, Array1, Array2};
+use zerovec::{ZeroMap, ZeroVec};
+
+#[derive(PartialEq, Debug, Clone, yoke::Yokeable, zerofrom::ZeroFrom)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize,databake::Bake),
+    databake(path = icu_segmenter),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct LstmMatrix<'data> {
+    #[serde(borrow)]
+    pub dim: ZeroVec<'data, i16>,
+    #[serde(borrow)]
+    pub data: ZeroVec<'data, f32>,
+}
 
-/// 'LstmData' is a struct that store a LSTM model. Its attributes are:
+impl<'data> LstmMatrix<'data> {
+    pub fn as_ndarray1(&self) -> Result<Array1<f32>, Error> {
+        if self.dim.len() == 1 {
+            Ok(Array::from_vec(self.data.to_vec()))
```

Comment:
Yeah, `self.data.to_vec()` reads the whole thing and puts it into new, aligned memory. I think this is okay for the time being because it gets LSTM unblocked. There is room to improve this in future quarters by introducing a new version of the key with better data types.